compiler/rustc_middle/src/middle/stability.rs

```diff
@@ -292,6 +303,32 @@ fn skip_stability_check_due_to_privacy(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
     }
 }
 
+// See issue #83250.
+fn get_sugg_for_allocator_api(
+    tcx: TyCtxt<'_>,
+    def_id: DefId,
+    span: Span,
+    feature: Symbol,
+) -> Option<(Span, String, String, Applicability)> {
+    if feature == Symbol::intern("allocator_api") {
```

Comment:
You should be able to pre-intern this string by adding it to this macro:

https://github.com/rust-lang/rust/blob/a2a7683e8f321e3c0b9d648ae480827b5ab70e1d/compiler/rustc_span/src/symbol.rs#L124

It should then be available as `sym::allocator_api` (or something like that).